### PR TITLE
Support .d.ts style enum generation

### PR DESCRIPTION
**Pull request by yoavsion**
This change replaces `enableConstEnums` and `enableEnumTypes` with a single cli argument – `enumGenType`, which supports the following formats:

**default**
```ts
export enum SomeEnum {
  Key1 = 'value1',
  Key2 = 'value2',
}
```

**const**
```ts
export const enum SomeEnum {
  Key1 = 'value1',
  Key2 = 'value2',
}
```

**literal**
```ts
export const SomeEnum  = {
  Key1: 'value1',
  Key2: 'value2',
} as const;
export type SomeEnum = typeof SomeEnum[keyof typeof SomeEnum];
```

**dts**
```ts
export const SomeEnum: {
  Key1: 'value1',
  Key2: 'value2',
};
export type SomeEnum = typeof SomeEnum[keyof typeof SomeEnum];
```

### CODE REVIEW

File: test/__snapshots__/test/test.ts.md [hunk 1]
```diff
@@ -7083,3 +7083,347 @@ Generated by [AVA](https://avajs.dev).
...
+      impliedHeterogeneousEnum?: -20.1 | null | "foo" | false;␊
+      namedIntegerEnumTitle: NamedInteger;␊
+      impliedNamedIntegerEnumTitle: ImpliedNamedInteger;␊
+      oneOfNamedEnum: IntegerOneOfNamedEnum | StringOneOfNamedEnum;␊
+      anyOfNamedEnum: IntegerAnyOfNamedEnum | StringAnyOfNamedEnum;␊
+      allOfNamedEnum: IntegerAllOfNamedEnum & StringAllOfNamedEnum;␊
+    }␊
+    ␊
+    export const NamedIntegerEnum: {␊
+      One: 1;␊
```
Comment (yoavsion) [inline test/__snapshots__/test/test.ts.md:7279]: The code actually generates these as `One: 1,` – but the prettier settings of this project change this to `One: 1;`.

File: src/index.ts [hunk 2]
```diff
@@ -15,6 +15,27 @@ import {validate} from './validator'
 
 export {EnumJSONSchema, JSONSchema, NamedEnumJSONSchema, CustomTypeJSONSchema} from './types/JSONSchema'
 
+/**
+ * Defines the available options for how enums can generated
+ */
+export const EnumGenType = {
+  Default: 'default',
+  /**
+   * Prepend enums with [`const`](https://www.typescriptlang.org/docs/handbook/enums.html#computed-and-constant-members)?
```
Comment (ajmacd) [inline src/index.ts]: I think you probably want to link to this?
https://www.typescriptlang.org/docs/handbook/enums.html#const-enums

**Review by ajmacd — COMMENTED** [review]
Looks good with a minor comment.

One thing I'm not quite clear on though: if you actually want to use the const object at runtime, you'll need a corresponding `.ts`/`.js` file containing a (redundant in some sense) assignment, right? Will the responsibility for generating that be outside this package?

File: src/index.ts (hunk 2 above)
Comment (yoavsion) [reply src/index.ts]: This was copied from the then-existing `enableConstEnums` flag documentation, but gladly updated! 🙌🏻

**Comment by yoavsion** [comment]
Yeah, I've been wondering the same; when integrating this change into webapp, I'll see whether that's needed 🙏🏻 

Also, realized I forgot to update the package version, one sec.

**Review by ajmacd — APPROVED** [review]

**Pull request merged**